ksql-serde/src/main/java/io/confluent/ksql/serde/avro/AvroDataTranslator.java

```diff
@@ -17,12 +17,14 @@
 
 import static java.util.Objects.requireNonNull;
```

Comment:
Also, we should set the connect avro precision property ("connect.decimal.precision") when converting to the connect schema, otherwise the converter will just pick a default precision for the avro schema. See https://github.com/confluentinc/schema-registry/blob/cdd227887945e70e2218c7beef1715f26daa9a3e/avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java#L822

Reply:
Great catch! I will update this

Reply:
The schema inference work is pretty straightforward so I added it here.